### PR TITLE
Secure python interop (plus Python<->Python interop unit tests)

### PR DESCRIPTION
**Pull request by nathanielmanistaatgoogle**
@soltanmm please review and merge.

### CODE REVIEW

File: src/python/src/grpc/_adapter/_channel.c [hunk 1]
```diff
@@ -42,19 +42,35 @@
 static int pygrpc_channel_init(Channel *self, PyObject *args, PyObject *kwds) {
   const char *hostport;
   PyObject *client_credentials;
-  static char *kwlist[] = {"hostport", "client_credentials", NULL};
+  char *server_host_override;
```
Comment (soltanmm) [inline src/python/src/grpc/_adapter/_channel.c]: PyArg_ParseTupleAndKeywords ostensibly doesn't touch the C variable if it is an optional argument and Python user code does not specify the value. Might want to initialize this to `NULL`.

File: tools/run_tests/python_tests.json [hunk 2]
```diff
@@ -13,6 +13,8 @@
   "grpc.framework.face.event_invocation_synchronous_event_service_test",
   "grpc.framework.face.future_invocation_asynchronous_event_service_test",
   "grpc.framework.foundation._later_test",
-  "grpc.framework.foundation._logging_pool_test"
+  "grpc.framework.foundation._logging_pool_test",
```
Comment (soltanmm) [inline tools/run_tests/python_tests.json]: Hmm - sorry about not noticing this PR before I sent in #965. Now there're some collisions. :-/

File: src/python/src/grpc/_adapter/_channel.c (hunk 1 above)
Comment (nathanielmanistaatgoogle) [inline src/python/src/grpc/_adapter/_channel.c]: Good call; done.

File: tools/run_tests/python_tests.json (hunk 2 above)
Comment (nathanielmanistaatgoogle) [inline tools/run_tests/python_tests.json]: I've resolved the collisions by recreating my commits on top of the current head-of-master. It wasn't any significant trouble. :-)

**Comment by nathanielmanistaatgoogle** [comment]
Please re-review but do not merge until tests have re-passed. :-)

**Comment by nathanielmanistaatgoogle** [comment]
Tests have passed; please merge. :-)

**Pull request merged**